pkg/vmprovider/providers/vsphere2/vmlifecycle/bootstrap.go

```diff
@@ -148,7 +157,14 @@ func getBootstrapArgs(
 			continue
 		}
 
-		if len(r.Nameservers) == 0 || (isTKG && len(r.SearchDomains) == 0) {
+		if len(bootstrapArgs.DNSServers) == 0 && len(r.Nameservers) == 0 {
+			missingDNSInfo = true
```

Comment:
Would you consider renaming `missingDNSInfo` to `getDNSInformationFromConfigMap`? It took me a bit to realize what this flag was guarding (I had to expand the code that did not change in order to see what `missingDNSInfo` did.